### ISSUE TITLE
Fix  `declaration-property-value-no-unknown` false negatives for math functions inside non-math functions

### ISSUE THREAD

**Issue by romainmenke** [stylelint/stylelint#9052]
### What minimal example or steps are needed to reproduce the bug?

```css
a {
  background: image-set(calc(2px + 1));
}
```

### What minimal configuration is needed to reproduce the bug?

```js
export default {
  rules: {
    "declaration-property-value-no-unknown": true
  }
};
```

### How did you run Stylelint?

[Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcICGACYAdAdilAjJYA1gOYBOEArplPCgJYC2SxMAtAM4wAuAFGEgBswPAEwAHAB4oA1CgCMASgUBuLDgAWMOsXVda-IX0HDxU2YpVYAviAA0IAGZ0BMAHJIGcRDAkexLgDowdnY7cAhMJ2IEEB8xCFIuFFgHJAoBJIxsFFJ0mHZaLJwcdBBYMAEkUiQuOgjWMXIxGESAT1YAN0EKNkwIVipCPoB3TFLaLlyYNRQra1VMMMhI7QAxBKYuGPYuVpcBOkwuIIiogIYAK1D7WDFQxCKUUp29mAOj8aeQSq58rlLbDNnrt9ocuKxllEOFwkNQqlBPqUfn9StYsCArEA)

### Which Stylelint-related dependencies are you using?

```json
{
  "stylelint": "17.2.0",
  "stylelint-config-standard": "40.0.0"
}
```

### What did you expect to happen?

I expected a warning for `calc()` inside non-math functions

### What actually happened?

No error was reported
